### PR TITLE
refactor(LayoutMenu): text wrapping support for section items and link

### PR DESCRIPTION
**Pull request by bndiep**
### Overview

The titles of nav items within the `LayoutMenu` currently do not wrap to the next line inside their containers. If title lengths are really long and beyond the width of its container, the text would overflow because we've set the width of `Anchor` within `SectionItemLink` to be "max-width." To ensure that we maintain focus states on tab _AND_ the section item/links wrap to the next line for the occasional long title, this change will set the widths to "fit-content" to allow text wrapping.

**Notes**
Before (with focus tabbing):
Screen Shot 2021-09-17 at 12 38 36 PM

After (with focus tabbing):
Screen Shot 2021-09-17 at 12 37 54 PM

- [ ] Related to designs:
- [ ] Related to JIRA ticket: ABC-123
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.84-alpha.24619c.0<br/>@codecademy/gamut-labs@20.1.1-alpha.24619c.0<br/>@codecademy/styleguide@44.6.7-alpha.24619c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6144e10c8f7fc69c4993f9da--gamut-preview.netlify.app>https://6144e10c8f7fc69c4993f9da--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6144e10c8f7fc69c4993f9da>Deploy Logs</a></p>

File: packages/gamut-labs/src/LayoutMenu/SectionItemLink.tsx [hunk 1]
```diff
@@ -15,7 +15,7 @@ export const SectionItemLink: React.FC<SectionItemLinkProps> = ({
     <Anchor
       variant="interface"
       display="block"
-      width="max-content"
+      width="fit-content"
```
Comment (codecaaron) [inline packages/gamut-labs/src/LayoutMenu/SectionItemLink.tsx:18]: We can probably just remove `width` props here if you want the text to wrap - `fit-content` has spotty coverage and wrapping comes for free with default behavior :)

Comment (bndiep) [reply packages/gamut-labs/src/LayoutMenu/SectionItemLink.tsx:18]: Ahhhh 🤩  even better! And it maintains the focus state on tab. Free is best :0

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.84-alpha.1b849c.0<br/>@codecademy/gamut-labs@20.1.1-alpha.1b849c.0<br/>@codecademy/styleguide@44.6.7-alpha.1b849c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6149098789acd2449fe6fba8--gamut-preview.netlify.app>https://6149098789acd2449fe6fba8--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6149098789acd2449fe6fba8>Deploy Logs</a></p>

**Review by codecaaron — APPROVED** [review]

**Review by aionate0812 — APPROVED** [review]
🚢

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.90-alpha.0d0f41.0<br/>@codecademy/gamut-labs@23.0.1-alpha.0d0f41.0<br/>@codecademy/styleguide@46.0.1-alpha.0d0f41.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://614b3d714a7cccc596befdc7--gamut-preview.netlify.app>https://614b3d714a7cccc596befdc7--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/614b3d714a7cccc596befdc7>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.91-alpha.7e3aa5.0<br/>@codecademy/gamut-labs@23.0.2-alpha.7e3aa5.0<br/>@codecademy/styleguide@46.0.2-alpha.7e3aa5.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://614c837f38bbd8bf0c3436f4--gamut-preview.netlify.app>https://614c837f38bbd8bf0c3436f4--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/614c837f38bbd8bf0c3436f4>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.91-alpha.077005.0<br/>@codecademy/gamut-labs@23.0.2-alpha.077005.0<br/>@codecademy/styleguide@46.1.1-alpha.077005.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://614e14836a459a167b8e969b--gamut-preview.netlify.app>https://614e14836a459a167b8e969b--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/614e14836a459a167b8e969b>Deploy Logs</a></p>

**Pull request merged**